File: Lab5/Lab5_CS.Task2/Letters.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab5_CS.Task2
{
    class Letters : Row
    {
        public Letters(string Row) : base(Row)
        {
            if (Row.Any(x => Char.IsDigit(x)))
                throw new Exception("Numbers in Letters constructor"); // checks, if there are numbers in string with letters.
        }
        public override int Length()
        {
            return base.Length();
        }
        public override void Remove(char ch = 'a') => _Row = new string(_Row.Where(c => !ch.ToString().Contains(c)).ToArray()); // removes char given in parameter list. If no parameter - removes 'a'.

    }
}

File: Lab5/Lab5_C++.Task2/Letters.cpp
#include "Letters.h"

Letters::Letters(string Row) : Row(Row) { 
	try
	{
		for (auto& i : Row)
			if (isdigit(i))
				throw exception("Digits in Letters constructor");
	}
	catch (const std::exception& ex)
	{
		cout << ex.what() << endl;
		throw ex;
	}
}

void Letters::Remove(char ch) {
	string temp;
	for (auto& i : _Row) {
		if (i == ch)
			continue;
		temp.push_back(i);
	}
	_Row = temp;
}
